src/Development/IDE/Core/Rules.hs

```diff
@@ -472,7 +480,7 @@ instance Hashable GhcSessionIO
 instance NFData   GhcSessionIO
 instance Binary   GhcSessionIO
 
-newtype GhcSessionFun = GhcSessionFun (FilePath -> Action HscEnvEq)
+newtype GhcSessionFun = GhcSessionFun (FilePath -> Action (IdeResult HscEnvEq))
```

Comment:
Notice I made `GhcSessionFun` return a proper `IdeResult` so that you can get proper diagnostics for failing cradle loads now.